packages/expo-print/ios/EXPrint/EXWKPDFRenderer.m

```diff
@@ -0,0 +1,92 @@
+
+@interface EXWKPDFRenderer () <WKNavigationDelegate>
+
+@property (nonatomic, assign) CGSize pageSize;
+@property (nonatomic, strong) WKWebView *webView;
+@property (nonatomic, strong) id<EXPDFRenderer> renderer;
+@property (nonatomic, strong) WKNavigation *htmlNavigation;
+@property (nonatomic, strong) void (^onRenderingFinished)(NSError * _Nullable, NSData * _Nullable, int);
+
+@end
+
+@implementation EXWKPDFRenderer
+
+- (void)PDFWithHtml:(NSString *)htmlString pageSize:(CGSize)pageSize completionHandler:(void (^)(NSError * _Nullable, NSData * _Nullable, int))handler
+{
+  _pageSize = pageSize;
+  _onRenderingFinished = handler;
+  _webView = [self createWebView];
+  if (@available(iOS 11.0, *)) {
+    _renderer = [[EXWKSnapshotPDFRenderer alloc] init];
+  } else {
+    _renderer = [[EXWKViewPDFRenderer alloc] init];
+  }
+  _htmlNavigation = [_webView loadHTMLString:htmlString baseURL:nil];
+}
+
+#pragma mark - UIWebViewDelegate
+
+- (void)webView:(WKWebView *)webView didFinishNavigation:(WKNavigation *)navigation
+{
+  if (_htmlNavigation != navigation) {
+    return;
+  }
+
+  UM_WEAKIFY(self);
+  [_renderer PDFFromWebView:webView completionHandler:^(NSError * _Nullable error, NSData * _Nullable data, int pagesCount) {
+    UM_ENSURE_STRONGIFY(self);
+    self.onRenderingFinished(error, data, pagesCount);
+  }];
+}
+
+- (void)webView:(WKWebView *)webView didFailNavigation:(WKNavigation *)navigation withError:(NSError *)error
+{
+  if (_htmlNavigation != navigation) {
+    return;
+  }
+
+  _onRenderingFinished(error, nil, 0);
+}
+
+- (WKWebView *)createWebView
+{
+  WKWebViewConfiguration *configuration = [[WKWebViewConfiguration alloc] init];
+//  configuration.userContentController = [self userContentController];
+  CGRect frame = CGRectMake(0, 0, _pageSize.width, _pageSize.height);
+  WKWebView *webView = [[WKWebView alloc] initWithFrame:frame configuration:configuration];
+  webView.navigationDelegate = self;
+  webView.backgroundColor = [UIColor clearColor];
+  webView.scrollView.showsHorizontalScrollIndicator = NO;
+  webView.scrollView.showsVerticalScrollIndicator = NO;
+
+  return webView;
+}
+
+- (WKUserContentController *)userContentController
+{
+  WKUserContentController *controller = [WKUserContentController new];
+  [controller addUserScript:[self scaleToOne]];
+  return controller;
+}
+
+- (WKUserScript *)scaleToOne
+{
+  NSString *addMetaScript = @""
+  "var meta = document.createElement('meta');"
+  "meta.setAttribute('name', 'viewport');"
+  "meta.setAttribute('content', 'width=device-width');"
+  "meta.setAttribute('initial-scale', '1.0');"
+  "meta.setAttribute('maximum-scale', '1.0');"
+  "meta.setAttribute('minimum-scale', '1.0');"
+  "meta.setAttribute('user-scalable', 'no');"
+  "document.getElementsByTagName('head')[0].appendChild(meta);";
+  return [[WKUserScript alloc] initWithSource:addMetaScript injectionTime:WKUserScriptInjectionTimeAtDocumentEnd forMainFrameOnly:YES];
```

Comment:
Shouldn't we put this at the document beginning? so people would be able to override this behavior

Reply:
Ah sorry, I forgot to remove this code altogether